modules/living_spaces_menus/living_spaces_menus.module

```diff
@@ -7,15 +7,30 @@
 
 use Drupal\Core\Template\Attribute;
 use Drupal\Core\Url;
+use Drupal\user\UserInterface;
+
+/**
+ * Implements hook_ENTITY_TYPE_update().
+ */
+function living_spaces_menus_user_update(UserInterface $user) {
+  $field_joined = 'joined_spaces';
+  $field_favorite = 'field_favorite_spaces';
+
+  if (($user->hasField($field_joined) && !$user->get($field_joined)->equals($user->original->get($field_joined)))
+    || $user->hasField($field_favorite) && !$user->get($field_favorite)->equals($user->original->get($field_favorite))) {
+    \Drupal::service('plugin.manager.menu.link')->rebuild();
+  }
+}
 
 /**
  * Implements hook_collapsible_menu_alter().
  */
 function living_spaces_menus_dom_bootstrap_collapsible_menu_alter(&$build) {
   $spaces_menu_key = 'menu_link_content:436e7a5e-7eb0-4ba7-928c-b9064865f174';
   $have_favorites = FALSE;
+  $groups = [];
 
-  $query = \Drupal::entityTypeManager()->getStorage('group')->getQuery();
+  $query = \Drupal::entityTypeManager()->getStorage('group')->getQuery()->accessCheck(FALSE);
```

Comment:
I guess we should check access for all spaces in the menu links. It's incorrect to show the space that will show 403 error after clicking on the link.

Please, merge the last changes. It already contains some fixes.

Reply:
Done!